plugins/vpp/model/l3/l3.proto

```diff
@@ -5,15 +5,26 @@ package l3;
 /* Static routes */
 message StaticRoutes {
     message Route {
-        uint32 vrf_id = 1;                   /* VRF identifier, field required for remote client. This value should be
+        enum RouteType {
+            INTRA_VRF = 0;                   /* Forwarding is being done in the specified vrf_id only, or according to
+                                                the specified outgoing interface. */
+            INTER_VRF = 1;                   /* Forwarding is being done by lookup into a different VRF,
+                                                specified as via_vrf_id field. In case of these routes, the outgoing
+                                                interface should not be specified. The next hop IP address
+                                                does not have to be specified either, in that case VPP does full
+                                                recursive lookup in the via_vrf_id VRF. */
+        }
+        RouteType type = 1;
```

Comment:
I don't think we should change the identification for fields. That's how some implementations for proto identify and decode the data.

Reply:
Makes sense if we only want to do backward-compatible changes in the proto files. This change makes it non-backward compatible. I can make it the member number 10 if you prefer that.

Do we have any guidelines on when we do only backward-compatible changes and when we are allowed to do non-backward compatible changes?

Reply:
Nothing written in repo AFAIK.. just what we discussed verbally maybe with other teams.